SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,374 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using SAMLPortal.Misc;
+using SAMLPortal.Models;
+using SAMLPortal.Models.Setup;
+using MySql.Data.MySqlClient;
+using Microsoft.EntityFrameworkCore;
+using CountryData.Standard;
+using Novell.Directory.Ldap;
+
+namespace SAMLPortal.Controllers
+{
+	[Route("Setup")]
+	[ServiceFilter(typeof(SetupAsyncActionFilter))]
+	public class SetupController : Controller
+	{
+		[HttpGet]
+		[Route("0")]
+		[Route("")]
+		public IActionResult Index()
+		{
+			return View();
+		}
+
+		[HttpGet]
+		[Route("1")]
+		public IActionResult StepOne()
+		{
+			var mysqlHost = Environment.GetEnvironmentVariable("SP_MYSQL_HOST");
+			var mysqlPortString = Environment.GetEnvironmentVariable("SP_MYSQL_PORT");
+			var mysqlDb = Environment.GetEnvironmentVariable("SP_MYSQL_DB");
+			var mysqlUser = Environment.GetEnvironmentVariable("SP_MYSQL_USER");
+
+			var mysqlPort = 3306;
+			if (mysqlPortString != null)
+			{
+				try
+				{
+					mysqlPort = Convert.ToInt32(mysqlPort);
+				}
+				catch (Exception)
+				{
+					// Not important if exception fires
+				}
+			}
+
+			StepOneModel model = new StepOneModel();
+			model.MySQLHost = mysqlHost != null ? mysqlHost : "";
+			model.MySQLPort = mysqlPort != 3306 ? mysqlPort : 3306;
+			model.MySQLDatabaseName = mysqlDb != null ? mysqlDb : "";
+			model.MySQLUser = mysqlUser != null ? mysqlUser : "";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("1")]
+		public IActionResult StepOne(StepOneModel model)
+		{
+			if (ModelState.IsValid && model != null)
+			{
+				try
+				{
+					string connectionString = "server=" + model.MySQLHost
+											+ ";port=" + model.MySQLPort.ToString()
+											+ ";database=" + model.MySQLDatabaseName
+											+ ";user=" + model.MySQLUser
+											+ ";password=" + model.MySQLPassword;
+
+					MySqlConnection connection = new MySqlConnection(connectionString);
+					connection.Open();
+
+					// If everything goes well
+					var fileName = GlobalSettings.Get("CONFIG_FILE");
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_HOST", model.MySQLHost);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_PORT", model.MySQLPort.ToString());
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_DB", model.MySQLDatabaseName);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_USER", model.MySQLUser);
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_MYSQL_PASS", model.MySQLPassword);
+
+					GlobalSettings.InitSettingsFromEnvironment();
+
+					SAMLPortalContext context = new SAMLPortalContext();
+					context.Database.Migrate();
+
+					// MySQL Step is complete
+					Helpers.ReplaceEnvVariableInFile(fileName, "SP_CONFIG_SETUPASSISTANT_STEP", "2");
+					GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "2");
+					return Redirect("2");
+				}
+				catch (Exception ex) when (ex is MySqlException)
+				{
+					if (ex is MySqlException)
+					{
+						Console.WriteLine(ex.Message);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)